### PR TITLE
feat(examples): add trivial grpc generation example

### LINKED ISSUE #2423: Cannot lookup runfile for @npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts
# 🐞 bug report

### Affected Rule

The generated tool rule for `@npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts`

### Is this a regression?

N/A (supposedly not, I don't know)

### Description

It seems there's no way to call to the `protoc-gen-ts` binary provided by `grpc_tools_node_protoc_ts`

## 🔬 Minimal Reproduction

Given the following packages in packages.json:

```json
"devDependencies": {
        "grpc-tools": "^1.10.0",
        "grpc_tools_node_protoc_ts": "^5.1.1"
}
```

The following rule will fail:

```python
genrule(
    name = "api_grpc",
    srcs = ["api.proto"],
    outs = [
      "api_grpc_pb.js",
      "api_pb.js",
      "api_grpc_pb.d.js",
      "api_pb.d.js",
    ],
    cmd = "./$(execpath @npm//grpc-tools/bin:grpc_tools_node_protoc) " +
      "-I`dirname $(<)` $(<) " +
      "--plugin=protoc-gen-ts=$(execpath @npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts) " +
      "--js_out=import_style=commonjs:`dirname $$(echo \"$(OUTS)\" | cut -f 1 -d ' ')`/ " +
      "--grpc_out=grpc_js:`dirname $$(echo \"$(OUTS)\" | cut -f 1 -d ' ')`/ " +
      "--ts_out=grpc_js:`dirname $$(echo \"$(OUTS)\" | cut -f 1 -d ' ')`/ ",
    tools = [
      "@npm//grpc-tools/bin:grpc_tools_node_protoc",
      "@npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts",
    ],
)
```

For any trivial `api.proto`.

## 🔥 Exception or Error

```
Error: Command failed: /private/var/tmp/_bazel_farcaller/7e2dc04345454e90edd8c97c601e6259/sandbox/darwin-sandbox/34/execroot/metanotes/bazel-out/host/bin/external/npm/grpc-tools/bin/grpc_tools_node_protoc.sh.runfiles/npm/node_modules/grpc-tools/bin/protoc --plugin=protoc-gen-grpc=/private/var/tmp/_bazel_farcaller/7e2dc04345454e90edd8c97c601e6259/sandbox/darwin-sandbox/34/execroot/metanotes/bazel-out/host/bin/external/npm/grpc-tools/bin/grpc_tools_node_protoc.sh.runfiles/npm/node_modules/grpc-tools/bin/grpc_node_plugin -Isrc/common/api src/common/api/api.proto --plugin=protoc-gen-ts=bazel-out/host/bin/external/npm/grpc_tools_node_protoc_ts/bin/protoc-gen-ts.sh --js_out=import_style=commonjs:bazel-out/darwin-fastbuild/bin/src/common/api/ --grpc_out=grpc_js:bazel-out/darwin-fastbuild/bin/src/common/api/ --ts_out=grpc_js:bazel-out/darwin-fastbuild/bin/src/common/api/
ERROR[runfiles.bash]: cannot look up runfile "npm/grpc_tools_node_protoc_ts/bin/protoc-gen-ts_require_patch.js"  (RUNFILES_DIR="./bazel-out/host/bin/external/npm/grpc-tools/bin/grpc_tools_node_protoc.sh.runfiles", RUNFILES_MANIFEST_FILE="")
ERROR[runfiles.bash]: cannot look up runfile "npm/grpc_tools_node_protoc_ts/bin/_protoc-gen-ts.module_mappings.json"  (RUNFILES_DIR="./bazel-out/host/bin/external/npm/grpc-tools/bin/grpc_tools_node_protoc.sh.runfiles", RUNFILES_MANIFEST_FILE="")
--ts_out: protoc-gen-ts: Plugin failed with status code 1.
```

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
OSX11.1
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Bazelisk version: v1.7.3
Build label: 3.6.0
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Oct 6 13:16:23 2020 (1601990183)
Build timestamp: 1601990183
Build timestamp as int: 1601990183
  </code>
</pre>

**Rules_nodejs version:**

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)

<pre>
  <code>
3.0.0
  </code>
</pre>

### PR DESCRIPTION
**Pull request by alexeagle**
Note that this doesn't use the code in our @bazel/labs package so maybe it's wrong?
Just trying to find a home for this answer https://github.com/bazelbuild/rules_nodejs/issues/2423#issuecomment-767881422
This uses https://www.npmjs.com/package/grpc_tools_node_protoc_ts but I know there are lots of alternative tools for grpc/proto codegen to .js/.d.ts

Fixes #2423

### CODE REVIEW

**Comment by farcaller** [comment]
FWIW I didn't even know `@bazel/labs` existed :-) It must have a really low visibility.

And yes, I actually stumbled on it now while browsing rules_nodejs examples for an unrelated issue.

File: examples/grpc/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,28 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "npm_package_bin")
+
+npm_package_bin(
+    name = "grpc",
+    outs = [
+        "api_grpc_pb.d.ts",
```
Comment (mrmeku) [inline examples/grpc/BUILD.bazel:6]: All these out files could be inffered from the name of proto if you were to make tiny macro wrapper

File: examples/grpc/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    args = [
+        "-I./" + package_name(),
+        "$(execpath api.proto)",
+        "--plugin=protoc-gen-ts=$(execpath @npm//:node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts)",
+        "--ts_out=$(@D)",
+        "--js_out=import_style=commonjs:$(@D)",
+        "--grpc_out=grpc_js:$(@D)",
+    ],
+    data = [
+        "api.proto",
```
Comment (mrmeku) [inline examples/grpc/BUILD.bazel:20]: I think you example should exhibit what happens when one proto depends upon another. The details there are probably important

**Comment by alexeagle** [comment]
I suspect we don't want to land this, instead there's some more canonical ruleset that gives a better grpc experience, like maybe contributing to https://github.com/rules-proto-grpc/rules_proto_grpc/pull/103

**Pull request closed without merging**